File: UsingTimberLogger/app/src/main/java/sample/github/nisrulz/usingtimberlogger/ReleaseTree.java
package sample.github.nisrulz.usingtimberlogger;

import android.util.Log;
import timber.log.Timber;

public class ReleaseTree extends Timber.Tree {

  @Override
  protected boolean isLoggable(String tag, int priority) {
    if (priority == Log.VERBOSE || priority == Log.DEBUG || priority == Log.INFO) {
      return false;
    }

    // Only log WARN, ERROR, WTF
    return true;
  }

  @Override
  protected void log(int priority, String tag, String message, Throwable t) {

    if (isLoggable(tag, priority)) {

      // Report all caught exceptions to crashlytics
      if (priority == Log.ERROR && t != null) {
        // Crashlytics.log(t)
      }
    }
  }
}

File: UsingTimberLogger/app/src/main/java/github/nisrulz/example/usingtimberlogger/ReleaseTree.kt
package github.nisrulz.example.usingtimberlogger

import android.util.Log
import timber.log.Timber

class ReleaseTree : Timber.Tree() {
    override fun isLoggable(tag: String?, priority: Int): Boolean {
        // Only log WARN, ERROR, WTF
        return !(priority == Log.VERBOSE || priority == Log.DEBUG || priority == Log.INFO)
    }

    override fun log(priority: Int, tag: String?, message: String, t: Throwable?) {
        if (isLoggable(tag, priority)) {

            // Report all caught exceptions to crashlytics
            if (priority == Log.ERROR && t != null) {
                // Crashlytics.log(t)
            }
        }
    }
}